src/library/scala/collection/immutable/List.scala

```diff
@@ -405,6 +405,90 @@ sealed abstract class List[+A] extends AbstractSeq[A]
   // Create a proxy for Java serialization that allows us to avoid mutation
   // during de-serialization.  This is the Serialization Proxy Pattern.
   protected final def writeReplace(): AnyRef = new List.SerializationProxy(this)
+
+  override def filter(p: A => Boolean): List[A] = filterImpl(p, isFlipped = false)
```

Comment:
Me neither, apparently it makes no diff. Sorry about that. The current filter is on AbstractTraversable. You always get both sigs, returning Object and List, when you define it on List, because that's what Self is.